### PR TITLE
Partial organization of mds/ header sections

### PR DESCRIPTION
**Pull request by batrick**
As we discussed in the CephFS summit, this is a simple organization of the class members in the mds/ headers following our C++ guidelines. The idea is to improve readability and understanding of each class' purpose.

I'll continue incrementally working on this as I find spare time/inclination.

### CODE REVIEW

File: src/mds/Capability.h [hunk 1]
```diff
@@ -132,11 +103,47 @@ class Capability {
...
+  static void *operator new(size_t num_bytes) {
+    void *n = pool.malloc();
+    if (!n)
+      throw std::bad_alloc();
+    return n;
+  }
+  void operator delete(void *p) {
+    pool.free(p);
+  }
+  const Capability& operator=(const Capability& other);  // no copying
```
Comment (gregsfortytwo) [inline src/mds/Capability.h]: Do the code style rules really require us to separate this and the copy constructor when disabling both? (Or does it perhaps only allow us to use better solutions?)

**Comment by gregsfortytwo** [comment]
Might as well run it through a test just in case though.

**Comment by dachary** [comment]
jenkins test this please (eio ignored in master)

**Comment by gregsfortytwo** [comment]
@jcsp, any thoughts before we merge this? (Don't want it to get stale!)

**Comment by jcsp** [comment]
No complaints here!

**Pull request merged**